### ISSUE TITLE
4k display

### ISSUE THREAD

**Issue by jrklasen** [SpiderOak/Encryptr#261]
great tool, I like it. However at my Linux Mint machine with a 4k display it is scaling badly.

**Comment by blue-bullet** [comment]
On 04/06/2016 11:48 PM, Jonas Klasen wrote:
>
> great tool, I like it. However at my Linux Mint machine with a 4k
> display it is scaling badly.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/SpiderOak/Encryptr/issues/261>
>
To scale is like a fish I do not eat.

**Comment by Scath-Enfys** [comment]
Are you opening in fullscreen or something? What do you expect with 4k...

**Comment by jrklasen** [comment]
It is independent of fullscreen or not.

I'm using it at a Linux mint 17.3 with Cinnamon desktop and a HiDPI monitor. The text and symbols from Encryptr are half the size than for other software, making it hart to read. At my desktop with the same configuration, but a non HiDPI monitor, everything looks normal.

**Comment by Scath-Enfys** [comment]
Since the project is written in mostly HTML and JS, have you noticed any suspicious font-size parameters?

**Comment by tilgovi** [comment]
This is a real problem. I appreciate the skepticism, but if you don't have experience running Linux in a high DPI environment, you should trust those of us reporting issues here.

Xrandr, GTK, QT ... these all have their own ways to affect scaling. It's also _not_ just about fonts. Fonts can usually be scaled, but getting menus and images and everything right takes more work.

It's a combination of understanding what the rendering engine supports and how it detects the environment. For a while, this was an issue with Chrome (WebKit/Blink) even when the desktop was properly set.

"Me too" and "check your fitness

**Comment by tilgovi** [comment]
Errr... sent too soon...


... "Me too" and "check your fonts" are both unhelpful at this point.

Someone just needs to sit down and sort through it and figure out what has to be changed. My guess is just a dependency upgrade.

**Comment by tilgovi** [comment]
If it's helpful, I can provide links for Xrandr, GTK, and QT settings so you can see the various hints the desktop environment might provide.

I can also link to issues from Gecko and Chromium.

I've carefully tracked HiDPI support for a number of projects since I got my current machine two years ago. I'm sure I can dig up links with little effort if it will help.

**Comment by tilgovi** [comment]
Also, Atom only recently solved this, too, after Chrome fixed it.

**Comment by ivan-novakov** [comment]
Running on Windows 10 on a 4K display - the SpideOak windows is has really tiny font no matter what the system settings are. Every other program is fine :).

**Comment by ysbaddaden** [comment]
This seems to be fixed in NW.js 0.16 for Linux:
https://github.com/nwjs/nw.js/issues/5165

**Comment by pacuna** [comment]
Any new on this? I'm using ubuntu gnome on my macbook retina and it would be great to be able to apply some scaling

**Comment by herrlado** [comment]
[URL=http://www.directupload.net/file/d/4646/6sct3kyv_png.htm][IMG]http://fs5.directupload.net/images/170228/temp/6sct3kyv.png[/IMG][/URL]

**Comment by santiagofn** [comment]
Some here using Win 10 on a Dell XPS. Fonts are ridicusly tiny with SpiderOak v. 6.1.9.
![spideroak with tiny fonts](https://cloud.githubusercontent.com/assets/8375708/26010810/6c409b08-371c-11e7-8630-93cb93fde6d2.png)

**Comment by anabelle** [comment]
I just installed the newer version of spideroak one and this is happening too.

**Comment by rstephe** [comment]
Same here.  As HiDPI and 4k screens are becoming commonplace now this is something SpiderOak needs to prioritize and fix quickly.  At a minimum, add a font-size preference to Preferences.

**Comment by chadfennell** [comment]
Yep, I'm on an XPS 13 9380 and having the same issue. I've seen this elsewhere with Electron apps like Spotify, which respond to `control +` to scale up the font size. The SpiderOak client doesn't present an obvious way to scale up the font.

**Comment by linwaytin** [comment]
I have the same issue in SpiderOakONE. 
The font is super small on a HiDPI screen.

**Comment by krulkrul** [comment]
i am also affected, running 20.04 Ubuntu

**Comment by pcdewolff** [comment]
So in 2022, 6 years after this thread started I installed a evaluation version of spideroak on a Windows 10 Computer with a 4K screen and I just can't read what is on my screen. Seems to be a very short evaluation.

**Comment by herrlado** [comment]
yes, i am planning to wait further 6 years, may be then we can us it...

**Comment by anabelle** [comment]
I decided to stick to my CRT 15" monitor while I wait so I can use it. 

![image](https://user-images.githubusercontent.com/445690/155035818-cbac3d8c-f0f4-416c-8500-10585efd29a0.png)

Who needs HD anyway?

**Comment by patmorin** [comment]
August 2022, and this is still happening on an XPS13 9320 with a 3840x2400 screen resolutino running Ubuntu 22.04.

**Comment by swyphcosmo** [comment]
I was just able to get SpiderOakONE scaled for my 4k screen on Ubuntu 22.10 using [run_scaled](https://github.com/kaueraal/run_scaled). I found the method described on [askubuntu](https://askubuntu.com/questions/960019/solution-for-apps-that-dont-scale-on-hidpi-displays).

After installing the dependencies, I copied `run_scaled` into my path. Created a new shell script, `SpiderOakONE_scaled` with the following contents:

```
#! /bin/sh
run_scaled SpiderOakONE
```

Then, I used [alacarte](https://manpages.ubuntu.com/manpages/kinetic/man1/alacarte.1.html) to update the SpiderOakONE shortcut to run `SpiderOakONE_scaled`. I also updated this in `Startup Applications`.

Now, I can finally read the settings!

Before:
![spideroakone_original](https://user-images.githubusercontent.com/7516713/199165418-ce53edbf-0d34-4b92-ae9d-4125903f0796.png)

After:
![spideroakone_scaled](https://user-images.githubusercontent.com/7516713/199165415-41eaf73c-ba8b-48da-afae-a328c708bd85.png)

**Comment by ziggutas** [comment]
February 2023 and same problem on Dell XPS13 with Pop!_OS.

SpiderOak support offer workaround advice for Windows: https://spideroak.support/hc/en-us/articles/115002046966-One-or-Groups-Appears-Very-Small-on-High-Resolution-Monitors.

Linux will have to wait: "Sorry, but we do not have a workaround for Linux. ONE is currently being updated from Python2 to Python3 which should resolve this issue, but I don't have an ETA for the release."

Tried the run_scaled method above but the process is just a bit too complicated for me and I could not set it up to run reliably without undesirable side effects.
